src/components/sidebar/routePathView/NewRoutePathView.tsx

```diff
@@ -4,98 +4,111 @@ import classnames from 'classnames';
 import RoutePathFactory from '~/factories/routePathFactory';
 import ViewHeader from '../ViewHeader';
 import RoutePathViewForm from './RoutePathViewForm';
 import * as s from './routePathView.scss';
 
 interface INewRoutePathViewProps {
     mapStore?: MapStore;
+    routeStore?: RouteStore;
     routePathStore?: RoutePathStore;
     networkStore?: NetworkStore;
 }
 
-@inject('mapStore', 'routePathStore', 'networkStore')
-@observer
-class NewRoutePathView extends React.Component<INewRoutePathViewProps>{
-    initialRoutePath: IRoutePath;
+interface INewRoutePathViewState {
+    currentRoutePath?: IRoutePath;
+}
 
+@inject('mapStore', 'routeStore', 'routePathStore', 'networkStore')
+@observer
+class NewRoutePathView extends React.Component<INewRoutePathViewProps, INewRoutePathViewState>{
     constructor(props: any) {
         super(props);
-        this.initialRoutePath = this.getInitialRoutePath();
+        this.state = {};
+    }
 
-         // TODO: call this when newRoutePath button is clicked
-        const newRoutePath = RoutePathFactory.createNewRoutePath('1', '1');
-        this.props.routePathStore!.setRoutePath(newRoutePath);
+    public onChange = (property: string) => (value: string) => {
+        this.setState({
+            currentRoutePath: { ...this.state.currentRoutePath!, [property]: value },
+        });
     }
 
-    // TODO: Use routePath store instead of state
-    // TODO, this is just a placeholder, implement real logic for creating new routePaths
-    private getInitialRoutePath = () => {
-        const newRoutePath: IRoutePath = {
-            internalId: '',
-            routeId: '',
-            routePathName: 'Uusi reitinsuunta',
-            routePathNameSw: 'Ny ruttriktning',
-            direction: '1',
-            positions: [[0, 0]],
-            geoJson: null,
-            visible: true,
-            startTime: new Date,
-            endTime: new Date,
-            lastModified: new Date,
-            routePathLinks: [],
-            originFi: '',
-            originSw: '',
-            destinationFi: '',
-            destinationSw: '',
-            routePathShortName: '',
-            routePathShortNameSw: '',
-            modifiedBy: '',
-        };
-        return newRoutePath;
+    public onSave = () => {
+        // TODO
     }
 
-    // TODO
-    public onChange = () => {
+    componentWillMount() {
+        if (!this.props.routePathStore!.routePath) {
+            this.initEmptyRoutePath();
+        }
     }
 
-    // TODO
-    public onSave = () => {
+    initEmptyRoutePath() {
+        const queryParams = navigator.getQueryParamValues();
+        const newRoutePath =
+            RoutePathFactory.createNewRoutePath(queryParams.lineId, queryParams.routeId);
 
+        this.props.routePathStore!.setRoutePath(newRoutePath);
     }
 
     componentDidMount() {
-        this.props.mapStore!.setNodeSize(NodeSize.large);
+        this.initStores();
+        this.initCurrentRoutePath();
```

Comment:
Do this already on constructor as discussed

Reply:
Done at constructor